apps/client/src/app/pages/portfolio/analysis/analysis-page.component.ts

```diff
@@ -201,13 +202,42 @@ export class AnalysisPageComponent implements OnDestroy, OnInit {
 
   public onChangeGroupBy(aMode: GroupBy) {
     this.mode = aMode;
+    this.fetchDividendsAndInvestments();
   }
 
   public ngOnDestroy() {
     this.unsubscribeSubject.next();
     this.unsubscribeSubject.complete();
   }
 
+  private fetchDividendsAndInvestments() {
```

Comment:
Created a separate method as this saves 2 api calls instead of calling `update()` as performance and positions do not change with the grouping mode